fecgov/fec-cms
# Report_type_full for AF canonical pages and datatable

## Summary (required)
- Display full name of the report on AF canonical pages using a dict added to constants based on list verified/edited by Rhiannon.
-  Rhiannon asked if this could be added to the AF datatable as well, so I added that to this PR.
- Show report code if full name not found in data/constants `report_type_full` dict
- Here is the latest list based on content/AF teams collaboration
https://docs.google.com/document/d/1QntwhFYqFYa6MG3q_hmJxm991ocAz2QNd5WbCvo0Vd8/edit

- [ ] **Possible followup work:** Should we update the report codes listed inn API docs based on the list in constants, some codes are wrong or missing. https://api.open.fec.gov/developers/#/filings/get_filings_

- Resolves #4912

### Required reviewers 
1 or 2 reviewers

(Include who is required to review prior to merge. For example: One designer and two front end developer reviews are required prior to merge)

## Impacted areas of the application

	modified:   data/constants.py
	modified:   legal/templates/legal-admin_fine.jinja
	modified:   legal/templates/partials/legal-search-results-afs.jinja
	modified:   legal/views.py
	
## How to test

This is on feature space
Example: https://fec-feature-cms.app.cloud.gov/data/legal/administrative-fine/4008/
AF Datatable: https://fec-feature-cms.app.cloud.gov/data/legal/search/admin_fines/

--OR--

- checkout and run branch
- See a AF canonical page like http://localhost:8000/data/legal/administrative-fine/4008/
- See http://127.0.0.1:8000/data/legal/search/admin_fines/ to see report name on datatable
- Confirm report name is listed and not report code
- To test the fallback,  you could change M10 in `data/constants.py` `report_type_full dict` to something else like MX to confirm that it reverts to showing the report code  if it does match  anything  in the constants dict.
